providers/provider_default.go

```diff
@@ -92,18 +92,28 @@ func (p *ProviderData) GetEmailAddress(_ context.Context, _ *sessions.SessionSta
 	return "", ErrNotImplemented
 }
 
-// ValidateGroup validates that the provided email exists in the configured provider
-// email group(s).
-func (p *ProviderData) ValidateGroup(_ string) bool {
-	return true
-}
-
 // EnrichSessionState is called after Redeem to allow providers to enrich session fields
 // such as User, Email, Groups with provider specific API calls.
 func (p *ProviderData) EnrichSessionState(_ context.Context, _ *sessions.SessionState) error {
 	return nil
 }
 
+// Authorize performs global authorization on an authenticated session.
```

Comment:
Do we need this changes in oidc.go as well.

Reply:
And one more issue is , I am getting x-auth-request-groups   with >50 groups  from oauth2-proxy , I would like to apply AuthorizationPolicy. on groups.    Istio does not support  *group1*. which only support  prefix and suffix match . How can we handle this as well.   Currenly I have handled very crude way, Is there any generic way we can handle this ?

Reply:
@NickMeves  Any idea how can this be handled ?  My whole point of getting **groups**  header is to apply istio  RBAC.  I understand solution has to come from istio.   Any suggestion would be helpful.

Reply:
I'm not an Istio user, so I'm not able to help troubleshoot.

But this sounds like maybe the problem being discussed here: https://github.com/oauth2-proxy/oauth2-proxy/pull/799